pulser/waveforms.py

```diff
@@ -192,38 +197,38 @@ def samples(self):
         return np.concatenate([wf.samples for wf in self._waveforms])
 
     @property
-    def first_value(self):
+    def first_value(self) -> float:
         """The first value in the waveform."""
         return self._waveforms[0].first_value
 
     @property
-    def last_value(self):
+    def last_value(self) -> float:
         """The last value in the waveform."""
         return self._waveforms[-1].last_value
 
     @property
-    def waveforms(self):
+    def waveforms(self) -> list:
```

Comment:
This one would be even better as:
```suggestion
    def waveforms(self) -> List[Waveform]:
```